draft-barnes-acme.md

```diff
@@ -218,304 +219,275 @@ In this section, we describe the four certificate management functions that ACME
+The remainder of this section provides the details of how these resources are structured and how the ACME protocol makes use of them.
 
+At the beginning of the ACME process, the client must be configured with URLs for the server's new-authorization and new-certificate resources.  The client learns URIs for any authorization and certificate resources it requires through the ACME protocol.
 
-## General Request/Response Lifecycle
+All ACME requests with a non-empty body MUST encapsulate the body in a JWS object, signed using the account key pair.  The server MUST verify the JWS before processing the request.  (For readability, however, the examples below omit this encapsulation.)  Encapsulating request bodies in JWS provides a simple authentication of requests by way of key continuity.
 
-Client-server interactions in ACME are logically request/response transactions, corresponding directly to HTTPS requests and responses.  The client sends a request message of a particular type, and the server sends response of a corresponding type.
+Note that this implies that GET requests are not authenticated.  For the most part, this is not an issue, since the ACME resources that can be accessed in this way (authorizations and certificates) usually do not contain sensitive information.  Servers that are concerned about parties other than the client gaining access to information via GET requests can create high-entropy authorization URLs and certificate URLs, so that they URLs themselves act as a shared secret between the client and server.
 
-All requests for a given ACME server are sent to the same HTTPS URI.  It is assumed that clients are configured with this URI out of band.  ACME requests MUST use the POST method, and since they carry JSON payloads, SHOULD set the Content-Type header field to "application/json".  ACME responses MUST be carried in HTTP responses with the status code 200.  ACME clients SHOULD follow HTTP redirects (301 or 302 responses), in case an ACME server is relocated.
+## Authorization Resources
 
-ACME provides three general message types -- "error", "defer", and "statusRequest" -- to cover cases where the server is not able to return a successful result immediately.  If there is a problem that prevents the request from succeeding, then the server sends an error message.  The fields in an error message are as follows:
+An ACME authorization resource represents server's authorization for an account key pair to represent an identifier.  In addition to a public key and identifier, an authorization includes several metadata fields, such as the status of the authorization (e.g., "pending", "valid", or "revoked") and which challenges were used to validate possession of the identifier.
```

Comment:
Related to my comment above: Do we consider authorizations to be confidential? For instance, some users may expect their new domain to remain confidential until they publish it. Since we're implementing Certificate Transparency, that is not a valid expectation. There's also a notion of private subdomains within CT, but I think we will probably not allow those either.

Does exposing the 'pending' status of a validation make it easier to attack that validation? Probably not, or we have a bug.

Since an authorization resource includes the public half of the account key, it does introduce a factorization risk that doesn't exist if we keep account public keys secret. It also introduces a risk of publicly tying together unrelated identifiers by an account key that the user may not have intended.

Action: I think we should just state explicitly that authorization resources are not confidential.